lib/routes.js

```diff
@@ -2,6 +2,40 @@
 
 const logger = require('./logger');
 const modbus = require('./modbus');
+const async = require('async');
+
+const queueDelay = 100;
+const queueMaxSize = 100;
+
+// Queue handler
+var q = async.queue(function (task, callback) {
+  logger.debug('🚥 Processing task: ' + task.name);
+
+  // Force callbacks to only execute one at a time
+  setTimeout(() => {
```

Comment:
This will not execute one at a time, it will just delay 100ms before executing each one. So they will still run at the same time, just 100ms into the future :). You don't want to be messing with `setTimeout` here, the `queue` handles the timing for you.

* You need your callback to return a Promise which doesn't resolve until the request has been finished. The simplest way to do this is to push `async function`s onto the queue and make sure they `await` everything they do.
* Then, here, you await the callback, so that the queue waits for the callback to finish.
* If you want an additional delay between handling requests you want `const timers = require('timers/promises')` at the top and `await timers.setTimeout(delay)` here. The old callback `setTimout` API doesn't do well with `async`/`await`.